animation/animation.ino

```diff
@@ -0,0 +1,394 @@
+    dir = 5; // increase brightenss
+  }
+  //Color change if button is released
+  if(is_button_pressed == false & last_buttonpress == true) {
+    global_hue = random8();
+    Serial.println(global_hue);
+  }
+  last_buttonpress = is_button_pressed;
+}
+
+///////////////////////////////////////
+// 6. Vibe Fireworks
+//// Upon press, flash hoops and bars at random times
+///////////////////////////////////////
+void vibe_fireworks(){
+
+static bool flashing = false;
+
+///green light to launch firework if probability condition is satisfied
+    if( random8() < chance_of_firework) {
+    go_firework = true;
+  }
+
+//If button is pressed for first time when animation is not flashing, then turn flashing on and log start of flashing animation
+if(is_button_pressed){
+  if(!flashing){
+    flashing_start_time = millis();
+    flashing = true;
+  }
+}
+// Flash firework if button is down and have green light to launch. Hold the flash for 1 second 
+    if(is_button_pressed && go_firework) {
+      if(millis() - flashing_start_time <= 1000){
+        for(int i = 0; i < NUM_LEDS; i++){
+		    uint8_t bri = 255;
+		    leds[i] = CHSV(global_hue, 200, bri);
+      }
+	}
+  //Fade out the firework after being shown for one second
+    else if(millis() - flashing_start_time >= 1000 && millis() - flashing_start_time <= 2000) {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(global_hue, 200, bri);
+      }
+      bri = bri - 5;
+    }
+    //turn lights off after 2 seconds have passed since firework launch
+    else if(millis() - flashing_start_time >= 2000) {
+     for(int i = 0; i < NUM_LEDS; i++){
+    leds[i] = CHSV(0, 0, 0);
+    }
+    flashing = false;
+    }
+    }
+    else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+    FastLED.show();
+  
+  //Color change if button is released
+  if(is_button_pressed == false & last_buttonpress == true) {
+    global_hue = random8();
+    Serial.println(global_hue);
+  }
+  last_buttonpress = is_button_pressed;
+  if(!flashing)
+    {go_firework = false;}
+}
+
+
+///////////////////////////////////////
+// 7. Vibe sparkle
+//// Upon press, vibe is lit and sparkles while held
+///////////////////////////////////////
+
+void vibe_sparkle() 
+{
+   if(is_button_pressed) {
+      for(int i = 0; i < NUM_LEDS; i++){
+		uint8_t bri = 255;
+		leds[i] = CHSV(global_hue, 200, bri);
+    addGlitter(10);
+	}
+  }
+    else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+    
+    FastLED.show();
+  
+  //Color change if button is released
+  if(is_button_pressed == false & last_buttonpress == true) {
+    global_hue = random8();
+    Serial.println(global_hue);
+  }
+  last_buttonpress = is_button_pressed;
+}
+
+void addGlitter( fract8 chanceOfGlitter) 
+{
+  if( random8() < chanceOfGlitter) {
+    leds[ random16(NUM_LEDS) ] += CRGB::White;
+  }
+}
```

Comment:
Might be helpful to add a reference to where this code came from so you can find it later